modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/RoundData.scala

```diff
@@ -44,9 +54,22 @@ case class RoundData(
   def addPeerCancellation(cancellation: CancelledBlockCreationRound): RoundData =
     this.focus(_.peerCancellations).modify(_ + (cancellation.senderId -> cancellation.reason))
 
-  def formBlock(): Option[DAGBlock] =
-    NonEmptySet
-      .fromSet((ownProposal.transactions ++ peerProposals.values.flatMap(_.transactions).toSet).toSortedSet)
-      .map(DAGBlock(tips.value, _))
+  def formBlock[F[_]: Async: KryoSerializer: Logger](validator: TransactionValidator[F]): F[Option[DAGBlock]] =
+    (ownProposal.transactions ++ peerProposals.values.flatMap(_.transactions)).toList
+      .traverse(validator.validate)
+      .flatMap { validatedTxs =>
+        val invalid = validatedTxs.collect { case Invalid(e) => e }
+        val valid = validatedTxs.collect { case Valid(tx) => tx }
 
```

Comment:
What about this: `val (invalid, valid) = validatedTxs.partitionMap(_.toEither)` ?